code/drasil-lang/lib/Language/Drasil/Expr/Lang.hs

```diff
@@ -206,6 +216,17 @@ assocArithOperToTy MulI  = S.Integer
 assocArithOperToTy AddRe = S.Real
 assocArithOperToTy MulRe = S.Real
 
+-- helper function for typechecking to help reduce duplication
+vvvInfer :: TypingContext Space -> VVVBinOp -> Expr -> Expr -> Either Space TypeError
+vvvInfer ctx op l r = case (infer ctx l, infer ctx r) of
+    (Left lt@(S.Vect lsp), Left (S.Vect rsp)) -> if lsp == rsp && S.isBasicNumSpace lsp
+                                                      && lsp /= S.Natural
```

Comment:
Naturals are always an interesting case for us. We can allow vector addition of naturals, but I think vector cross product of naturals is a bit dubious, same as subtraction (which would really be monus).

This is the only thing there is to _really_ question (everything else seems spotless).

Reply:
Agreed. Done in next check-in.